File: src/main/java/net/internetisalie/lunar/LuaFileType.java
package net.internetisalie.lunar;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class LuaFileType extends LanguageFileType {
    public static final LuaFileType INSTANCE = new LuaFileType();

    private LuaFileType() {
        super(LuaLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Lua File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Lua language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "lua";
    }

    @Override
    public Icon getIcon() {
        return LuaIcons.FILE;
    }

}

File: src/main/kotlin/net/internetisalie/lunar/lang/LuaFileType.kt
package net.internetisalie.lunar.lang

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object LuaFileType : LanguageFileType(LuaLanguage) {
    override fun getName(): String = "Lua"

    override fun getDescription(): String = "Lua language file"

    override fun getDefaultExtension(): String = "lua"

    override fun getIcon(): Icon = LuaIcons.FILE
}
